src/cascadia/TerminalSettingsModel/KeyChordSerialization.cpp

```diff
@@ -5,96 +5,108 @@
+    XX(VK_F3, L"f3")                                     \
+    XX(VK_F4, L"f4")                                     \
+    XX(VK_F5, L"f5")                                     \
+    XX(VK_F6, L"f6")                                     \
+    XX(VK_F7, L"f7")                                     \
+    XX(VK_F8, L"f8")                                     \
+    XX(VK_F9, L"f9")                                     \
+    XX(VK_F10, L"f10")                                   \
+    XX(VK_F11, L"f11")                                   \
+    XX(VK_F12, L"f12")                                   \
+    XX(VK_F13, L"f13")                                   \
+    XX(VK_F14, L"f14")                                   \
+    XX(VK_F15, L"f15")                                   \
+    XX(VK_F16, L"f16")                                   \
+    XX(VK_F17, L"f17")                                   \
+    XX(VK_F18, L"f18")                                   \
+    XX(VK_F19, L"f19")                                   \
+    XX(VK_F20, L"f20")                                   \
+    XX(VK_F21, L"f21")                                   \
+    XX(VK_F22, L"f22")                                   \
+    XX(VK_F23, L"f23")                                   \
+    XX(VK_F24, L"f24")                                   \
+    XX(VK_ADD, L"numpad_plus", L"numpad_add")            \
+    XX(VK_SUBTRACT, L"numpad_minus", L"numpad_subtract") \
+    XX(VK_MULTIPLY, L"numpad_multiply")                  \
+    XX(VK_DIVIDE, L"numpad_divide")                      \
+    XX(VK_DECIMAL, L"numpad_period", L"numpad_decimal")  \
+    XX(VK_NUMPAD0, L"numpad0", L"numpad_0")              \
+    XX(VK_NUMPAD1, L"numpad1", L"numpad_1")              \
+    XX(VK_NUMPAD2, L"numpad2", L"numpad_2")              \
+    XX(VK_NUMPAD3, L"numpad3", L"numpad_3")              \
+    XX(VK_NUMPAD4, L"numpad4", L"numpad_4")              \
+    XX(VK_NUMPAD5, L"numpad5", L"numpad_5")              \
+    XX(VK_NUMPAD6, L"numpad6", L"numpad_6")              \
+    XX(VK_NUMPAD7, L"numpad7", L"numpad_7")              \
+    XX(VK_NUMPAD8, L"numpad8", L"numpad_8")              \
+    XX(VK_NUMPAD9, L"numpad9", L"numpad_9")              \
+    XX(VK_OEM_PLUS, L"plus")
 
-static constexpr int MAX_CHORD_PARTS = 5; // win+ctrl+alt+shift+key
+// clang-format off
+using nameToVkeyPair = std::pair<std::wstring_view, int32_t>;
+static const til::static_map nameToVkey{
+    // The above VKEY_NAME_PAIRS macro contains a list of key-binding names for each virtual key.
+    // This god-awful macro inverts VKEY_NAME_PAIRS and creates a static map of key-binding names to virtual keys.
+#define GENERATOR_1(vkey, name1) nameToVkeyPair{ name1, vkey },
+#define GENERATOR_2(vkey, name1, name2) nameToVkeyPair{ name1, vkey }, nameToVkeyPair{ name2, vkey },
+#define GENERATOR_3(vkey, name1, name2, name3) nameToVkeyPair{ name1, vkey }, nameToVkeyPair{ name2, vkey }, nameToVkeyPair{ name3, vkey },

```

Comment:
Curious: you don't _technically_ need `GENERATOR_3` right? Because we don't have any pairs in VKEY_NAME_PAIRS that are 1 vkey and 3 strings.

Reply:
Yep I just added it prematurely in case we add a third one at some point. It'll then prevent someone from becoming fairly confused. Also I thought adding at least 3 cases would be a minimum for someone unfamiliar with the code (and this variadic macro pattern) to recognize how to add a 4th one.